content/en/docs/reference/command-line-tools-reference/feature-gates/storage-namespace-index.md

```diff
@@ -0,0 +1,14 @@
+---
+title: StorageNamespaceIndex
+content_type: feature_gate
+_build:
+  list: never
+  render: false
+
+stages:
+  - stage: beta
+    defaultValue: true
+    fromVersion: "1.30"
+---
+Allows namespace indexer for namespace scoped resources in API server cache
+to accelerate list operations.
```

Comment:
I'm afraid I'd have to read the code to figure out what this means.

Reply:
Sadly, this is true. We have witnessed a tendency to feature gate abuse.
Whenever the developers are unsure whether a change is good, they add a new feature gate.
Sometimes the new "gate" is not about a user-facing feature. It is merely an
improvement to the existing behavior which may and may not be perceived by the users.
A gate is added anyway.
I'm really doubting whether there are people playing with this kind of gates.
I'm also concerning whether we as a community has a well established criteria for gates.
To an extreme end, we can add gate for whatever changes introduced. The reasons can be
that the requirements are unclear, the implementation is not solid, the effects are
yet to be evaluated, so on and so forth.
In an ideal world, feature gates should be constrained to changes that really make
a big difference for the users, changes that are understandable for users.
But, you know, the real life always sucks.

Reply:
@sftim @tengqm For the namespace indexer feature, we did a long time discussion and did a benchmark that shown excellent performance in terms of latency, CPU, and memory in large clusters (like with 100k+ pods) for listing pods within a single namespace with namespace indexers. This implementation significantly reduces cpu and memory consumption, especially when dealing with a small number of pods.
Discussion:
- https://github.com/kubernetes/kubernetes/issues/120778
Benchmark doc: 
- https://docs.google.com/document/d/1bTxGqlH0tqrpH16NxC2SIUTGMN3_1zNKRiRdIW9Y5VU/edit?usp=sharing

Reply:
Also, we are from eBay Cloud team and this feature already is applied to our production that runs good performance.

Reply:
Yes, just pods at the beginning from the recommendation @wojtek-t . If stable after running for some time, I would like to enable this for the major resources(We already enabled major resources in our internal kubernetes controlplane).